src/changelog/ticketChangelog.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Implements methods for working with ticket scopes, with the ability to search
+ * for common scopes both within the project and within the tags
+ */
+
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+import * as RA from "fp-ts/ReadonlyArray";
+import { pipe } from "fp-ts/lib/function";
+import { GenericTicket } from "../types";
+import { popFromRegex, isSameScope } from "../utils/validator";
+import { Scope, projectToScope, tagToScope } from "./types";
+
+export declare function warn(message: string): void;
+export declare function markdown(message: string): void;
+export declare function schedule<T>(asyncFunction: Promise<T>): void;
+
+// pattern used to recognize a scope label
+const regex = /(changelog-scope:|epic-)(.*)/m;
+
+/**
+ *Try to detect the {@link Scope} of the ticket by projectId
+ */
+export const getProjectScope = (
+  ticket: GenericTicket
+): E.Either<Error, Scope> =>
+  pipe(
+    O.fromNullable(projectToScope[ticket.projectId]),
```

Comment:
`projectToScope` seems like a singleton or a global variable ⚠️

 It must be passed as a parameter to the function; otherwise, it is not pure.
 
Furthermore, if `projectToScope` is a `Record`, you can use the lookup function from `fp-ts/Record` to simplify the code.